qadence/backends/pyqtorch/convert_ops.py

```diff
@@ -286,17 +300,35 @@ def convert_block(
                 qubit_support[:-1],
                 qubit_support[-1],
                 extract_parameter(block, config),
-                noise=block.noise,
+                noise=convert_digital_noise(block.noise),
             )
         else:
             if "CSWAP" in block_name:
-                op = pyq_cls(qubit_support[:-2], qubit_support[-2:], noise=block.noise)  # type: ignore [attr-defined]
+                op = pyq_cls(
+                    qubit_support[:-2], qubit_support[-2:], noise=convert_digital_noise(block.noise)  # type: ignore [attr-defined]
+                )
             else:
-                op = pyq_cls(qubit_support[:-1], qubit_support[-1], noise=block.noise)  # type: ignore [attr-defined]
+                op = pyq_cls(
+                    qubit_support[:-1], qubit_support[-1], noise=convert_digital_noise(block.noise)  # type: ignore [attr-defined]
+                )
         return [op]
     else:
         raise NotImplementedError(
             f"Non supported operation of type {type(block)}. "
             "In case you are trying to run an `AnalogBlock`, make sure you "
             "specify the `device_specs` in your `Register` first."
         )
+
+
+def convert_digital_noise(noise: NoiseHandler | None) -> pyq.noise.NoiseProtocol | None:
+    if noise is None:
```

Comment:
Maybe do this check outside this function ? So that you don't have to do it every time it is called and you won't have to handle the `None` return type ?